protocols/raft/src/main/java/io/atomix/protocols/raft/zeebe/ZeebeLogAppender.java

```diff
@@ -28,8 +32,53 @@
    * @param lowestPosition lowest record position in the data buffer
    * @param highestPosition highest record position in the data buffer
    * @param data data to store in the entry
-   * @return a future which is completed with the indexed entry is committed
    */
-  CompletableFuture<Indexed<ZeebeEntry>> appendEntry(
-      long lowestPosition, long highestPosition, ByteBuffer data);
+  void appendEntry(long lowestPosition, long highestPosition, ByteBuffer data, Listener listener);
+
+  /**
+   * An append listener can observe and be notified of different events related to the append
```

Comment:
Maybe call it then AppendListener :smile:

Reply:
:+1: